src/frontend/src/common/analytics/eventTypes.ts

```diff
@@ -4,11 +4,168 @@
  * This exists to ensure consistency/clarity across types and to provide
  * a reference for data analysts working on CZ GE.
  *
- * - Each event type should have a plain-English description of what it means.
+ * --- REQUIREMENTS for adding new event types ---
```

Comment:
I like the idea of having the requirements outlined in english so new peeps can follow along. Do you feel these requirements are also codified, such that a new person who skips the comments can't make the mistake (ie: programmatic enforcement)?

Reply:
Some are, some aren't. Flat event structures are enforced, but things like naming and commenting what an event is/contains are not. I don't really have any ideas for I could enforce the naming/documentation aspect beyond some sort of extra tool that is running on the raw text of the file, and that feels bad for other reasons. I'm mostly pinning this on hope that A) devs do the same as they see already done in a file, B) PR reviews, and C) devs read the docs.

Reply:
It would be nice to have this documented somewhere in a readme page for frontend how-tos (and link to that in the code), I worry about the integrity of big comment blocks over time

Reply:
I'm happy to point it out or duplicate it in the readme, but I actually have a differing opinion. I agree that comment blocks getting out of sync with reality is a common issue, but I personally think documentation located in separate places drifts even faster. I think a giant comment block right next to the code in question has a (slightly) better chance of getting updated than a readme, simply because the dev doing the change might not be aware the readme exists or documents that stuff.